src/test/scala/cognite/spark/v1/wdl/WDLTrajectoriesTest.scala

```diff
@@ -0,0 +1,181 @@
+      |   t.doglegSeverityUnit,
+      |   t.phase,
+      |   r.rows as rows
+      | from wdl_test_raw_trajectories as t
+      | left join r on r.wellboreAssetExternalId = t.wellboreAssetExternalId
+      |""".stripMargin
+
+  it should "ingest and read bent trajectory" in {
+    val testIngestionsDF = spark
+      .sql(s"""$getTrajectoryRowsCTE
+           | $getTrajectoriesExpr
+           |""".stripMargin)
+      .filter("""wellboreAssetExternalId = "A:wb1"""")
+
+    testIngestionsDF.write
+      .format("cognite.spark.v1")
+      .option("project", "jetfiretest2")
+      .option("type", "welldatalayer")
+      .option("wdlDataType", "TrajectoryIngestion")
+      .useOIDCWrite
+      .save()
+
+    val expectedTrajectoriesDF = spark.read
+      .schema(testClient.getSchema("Trajectory"))
+      .json("src/test/resources/wdl-test-expected-trajectories.jsonl")
+      .filter("""wellboreAssetExternalId = "A:wb1"""")
+
+    val trajectoriesDF = sparkReader
+      .option("wdlDataType", "Trajectory")
+      .load()
+
+    (expectedTrajectoriesDF.collect() should contain).theSameElementsAs(trajectoriesDF.collect())
+  }
+
+  it should "ingest and read trajectory in feet" in {
+    val testIngestionsDF = spark
+      .sql(s"""$getTrajectoryRowsCTE
+           | $getTrajectoriesExpr
+           |""".stripMargin)
+      .filter("""wellboreAssetExternalId = "A:wb2"""")
+
+    testIngestionsDF.write
+      .format("cognite.spark.v1")
+      .option("project", "jetfiretest2")
+      .option("type", "welldatalayer")
+      .option("wdlDataType", "TrajectoryIngestion")
+      .useOIDCWrite
+      .save()
+
+    val expectedTrajectoriesDF = spark.read
+      .schema(testClient.getSchema("Trajectory"))
+      .json("src/test/resources/wdl-test-expected-trajectories.jsonl")
+      .filter("""wellboreAssetExternalId = "A:wb2"""")
+
+    val trajectoriesDF = sparkReader
+      .option("wdlDataType", "Trajectory")
+      .load()
+
+    (expectedTrajectoriesDF.collect() should contain).theSameElementsAs(trajectoriesDF.collect())
+  }
+
+  it should "fail while ingesting trajectory without rows" in {
+    val testIngestionsDF = spark
+      .sql("""select
+          |   t.wellboreAssetExternalId,
+          |   struct(t.sourceName, t.externalId as sequenceExternalId) as source,
+          |   t.type,
+          |   t.measuredDepthUnit,
+          |   t.inclinationUnit,
+          |   t.azimuthUnit,
+          |   t.doglegSeverityUnit,
+          |   t.phase,
+          |   array() as rows
+          | from wdl_test_raw_trajectories as t
+          |""".stripMargin)
+      .filter("""wellboreAssetExternalId = "A:wb1"""")
+
+    try {
+      testIngestionsDF.write
+        .format("cognite.spark.v1")
+        .option("project", "jetfiretest2")
+        .option("type", "welldatalayer")
+        .option("wdlDataType", "TrajectoryIngestion")
+        .useOIDCWrite
+        .save()
+    } catch {
+      case _: org.apache.spark.SparkException => ()
```

Comment:
Do we expect this to throw a SparkException? If so you can just do this instead 😁 
`a[SparkException] should be thrownBy ...`

Reply:
Great catch!
Leftover of debugging.